src/main.rs

```diff
@@ -0,0 +1,200 @@
+// (De-)Serialization
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
+extern crate serde_json;
+
+// HTTP
+extern crate actix_web;
+extern crate futures;
+#[macro_use]
+extern crate hyper;
+extern crate reqwest;
+
+// Logging and CLI
+extern crate chrono;
+extern crate fern;
+#[macro_use]
+extern crate log;
+
+use std::collections::HashMap;
+
+use actix_web::{http, server, App, AsyncResponder, Error, HttpMessage, HttpRequest, HttpResponse};
+use actix_web::middleware::Logger;
+
+use futures::future::Future;
+
+header! { (XCachetToken, "X-Cachet-Token") => [String] }
+
+fn hook(req: HttpRequest) -> Box<Future<Item = HttpResponse, Error = Error>> {
+    req.clone().json()                          // <- get JsonBody future
+        .from_err()
+        .and_then(move |alert: AlertHook| {  // <- deserialized value
+            info!("{:?}", alert);
+
+            //create map for json of cachet api call
+            let mut map = HashMap::new();
+            map.insert(
+                "status",
+                match alert.status {
+                    Status::Firing => alert.alerts[0].annotations.severity,
+                    Status::Resolved => 1,
+                },
+            );
+
+            //send http put to cachet
+            let client = reqwest::Client::new();
+            let response = match client.put(&format!(
+                    "{endpoint}/api/v1/components/{component}",
+                    endpoint = match std::env::var("CACHET_BASE_URL") {
+                        Ok(val) => val,
+                        Err(_) => String::from(include_str!("cachet_endpoint.txt")),
+                    },
+                    component = alert.alerts[0].annotations.component
+                ))
+                // read "Authorization: Bearer" token into "X-Cachet_Token"
+                .header(XCachetToken(match req.headers().get("Authorization") {
+                    Some(header) => match header.to_str() {
+                        Ok(header) => {
+                            let mut header_prefix = String::from(header);
+                            header_prefix.split_off(7) //cuts of the first 7 chars: "Bearer "
+                        },
+                        Err(_) => return Ok(HttpResponse::new(http::StatusCode::UNAUTHORIZED)),
```

Comment:
Do we want to log something like "Unable to find Bearer token in Authentication header"